crates/wasi-threads/src/lib.rs

```diff
@@ -27,6 +27,12 @@ impl<T: Clone + Send + 'static> WasiThreadsCtx<T> {
                 WASI_ENTRY_POINT
             );
         }
+        if let Err(err) = linker.instantiate_pre(&module) {
+            bail!(
+                "wasi-threads will not be able to instantiate the passed module: {:?}",
+                err
+            );
+        }
         Ok(Self { module, linker })
```

Comment:
Thinking about this again, the `WasiThreadsCtx` could actually store `InstancePre<T>` and use that to instantiate as there's no need to recreate it each time a thread is spawned.

Could `::new` be refactored to either:

* Take a `&Module` and `&Linker<T>` and persist `Arc<InstancePre<T>>` within `WasiThreadsCtx`
* Or take a `InstancePre<T>` directly and store an `Arc` internally in the context